src/bitcoin-tx.cpp

```diff
@@ -669,6 +669,10 @@ static void MutateTxSign(CMutableTransaction& tx, const std::string& flagStr)
         if (!fHashSingle || (i < mergedTx.vout.size()))
             ProduceSignature(keystore, MutableTransactionSignatureCreator(&mergedTx, i, amount, nHashType), prevPubKey, sigdata);
 
+        if (amount == MAX_MONEY && !sigdata.scriptWitness.IsNull()) {
+            throw std::runtime_error(strprintf("Missing amount for %s", coin.out.ToString()));
```

Comment:
I tend to agree that seeing `CTxOut(nValue=21000000.00000000...` when the error is for a missing amount is confusing. 

Using `MAX_AMOUNT` for the check is already a nuanced concept (I had to read through all the linked PRs/issues before I understood why it was being used here). I don't think displaying 0 is correct, either, since a 0 amount UTXO is a valid spend.

Perhaps just print the `scriptPubKey`? Something like `Missing amount for CTxOut with scriptPubKey=0014751e76e8199196d454941c45d1`

Reply:
Thanks, I agree that the best would be to not display the amount at all if it is missing and picked up your suggestion.